wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -193,6 +193,7 @@ class KotlinGenerator private constructor(
             newName("reader", "reader")
             newName("Builder", "Builder")
             newName("builder", "builder")
+            newName("readerTag", "readerTag")
```

Comment:
This name is not used as part of public API so it doesn't need to be part of these name allocators which track public API and change properties (as shown by the test).

Since the name is local to a method, you can clone the allocator inside `decodeFun` and call `newName` on the cloned allocator. This will avoid collision with the API names while not leaking the local name to the global name allocator.